### PR TITLE
certificates: use generic resource

### PR DESCRIPTION
**Pull request by mig4**
Towards https://github.com/giantswarm/giantswarm/issues/11392.

### CODE REVIEW

File: service/controller/resource/certificates/resource.go [hunk 1]
```diff
@@ -21,59 +23,82 @@ type Config struct {
...
 
-	return r, nil
-}
+	sc := secretCopier{clientFunc: clientFunc}
+
+	c := generic.Config{
+		ClientFunc:     clientFunc,
+		Logger:         config.Logger,
+		Name:           Name,
+		ToCR:           sc.ToCR,
```
Comment (TheoBrigitte) [inline service/controller/resource/certificates/resource.go:51]: Nice!

Comment (mig4) [reply service/controller/resource/certificates/resource.go:51]: I just had a thought that `ToCR` could also be a closure defined inside `New` so there wouldn't be a need for the struct, not sure if it's worth changing though.

Comment (TheoBrigitte) [reply service/controller/resource/certificates/resource.go:51]: True, first I wondered what was this struct for. Maybe its simpler with the closure, I have no strong opinion on this.

**Comment by mig4** [comment]
Tested on _godsmack_ BTW, all good.

**Pull request merged**